qutip/solver/nm_mcsolve.py

```diff
@@ -25,9 +29,21 @@
 #     https://arxiv.org/abs/2209.08958
 
 
-def nm_mcsolve(H, state, tlist, ops_and_rates=(), e_ops=None, ntraj=500, *,
-               args=None, options=None, seeds=None, target_tol=None,
-               timeout=None):
+def nm_mcsolve(
+    H: QobjEvoLike,
+    state: Qobj,
+    tlist: ArrayLike,
+    ops_and_rates: list[tuple[Qobj, float | CoefficientLike]] = (),
+    e_ops: EopsLike | list[EopsLike] | dict[Any, EopsLike] = None,
+    ntraj: int = 500,
+    *,
+    args: dict[str, Any] = None,
```

Comment:
Should `args` be removed here too?

Reply:
No, the `??solver` functions take operators as list, but the class interface takes QobjEvo. It does not have a `tlist` needed for some coefficient, it's also a lot cleaner.